gapidapk/deviceinfo.go

```diff
@@ -172,8 +174,16 @@ func preparePerfettoProducerLauncherFromApk(ctx context.Context, d adb.Device) (
 		return "", cleanup.Invoke(ctx), log.Errf(ctx, err, "Failed to unzip %v from %v", perfettoProducerLauncher, packageName)
 	}
 
-	// Finally, make sure the binary is executable
+	// Make sure the binary is executable
 	d.Shell("chmod", "a+x", launcherPath).Call(ctx)
+
+	// Finally, if there's an existing instance running, kill it.
+	if res, err := d.Shell("cat", perfettoProducerLauncherPidFile).Call(ctx); err == nil {
+		if _, err := strconv.Atoi(res); err == nil {
+			d.Shell("kill -2", res).Call(ctx)
```

Comment:
GAPID never deletes that file. What if the producer is stopped, then another process gets the same PID: we kill this other process?
Would there be a way, maybe using `ps`, to check that this PID is indeed a perfetto producer launcher?

Reply:
hmmmmm, let me think about it. I think maybe we can check whether there are producers registered from querying perfetto. Will explore it